### PR TITLE
Moves the bulk of shield code from the riot subtype into the core shield.

### PR DESCRIPTION
**Pull request by GuillaumePrata**
## About The Pull Request
I wanted to "rework" the riot shield security has but most other shields are subtypes of it, making them all inherit those features.
So I made this PR first to make my job easier and fix some odd bugs like fixing wooden bucklers by slapping them with titanium sheets...

There *shouldn't* be any other big change outside of the titanium healing and baton bashing they all inherited but I have been staring at shield code and worked on another branch before deciding to split the code improvement/fixes from the features so I might have missed something.

And while there, kill some single letter vars, remove unnecessary lines, etc.
## Why It's Good For The Game
Fixing a wooden shield by slapping it with titanium is weird... so is fixing a glass+iron shield but that is a bigger balance change.

And well, easier for future shield changes to not inherit weird behaviors from the riot shield.
## Changelog
:cl: Guillaume Prata
fix: You can no longer repair wooden bucklers and roman shields by slapping them with a titanium sheet.
/:cl:

### CODE REVIEW

File: code/game/objects/items/shields.dm [hunk 1]
```diff
@@ -1,32 +1,19 @@
...
 	throw_range = 3
 	w_class = WEIGHT_CLASS_BULKY
-	custom_materials = list(/datum/material/glass=7500, /datum/material/iron=1000)
 	attack_verb_continuous = list("shoves", "bashes")
 	attack_verb_simple = list("shove", "bash")
-	var/cooldown = 0 //shield bash cooldown. based on world.time
-	transparent = TRUE
-	max_integrity = 75
+	armor = list(MELEE = 50, BULLET = 50, LASER = 50, ENERGY = 0, BOMB = 30, BIO = 0, FIRE = 80, ACID = 70)
+	var/transparent = FALSE // makes beam projectiles pass through the shield
```
Comment (Fikou) [inline code/game/objects/items/shields.dm:16]: autodoc

File: code/game/objects/items/shields.dm [hunk 2]
```diff
@@ -67,58 +37,79 @@
 		if(0 to 25)
 			. += span_warning("It's falling apart!")
 
-/obj/item/shield/riot/proc/shatter(mob/living/carbon/human/owner)
-	playsound(owner, 'sound/effects/glassbr3.ogg', 100)
-	new /obj/item/shard((get_turf(src)))
+/obj/item/shield/proc/shatter(mob/living/carbon/human/owner)
```
Comment (Fikou) [inline code/game/objects/items/shields.dm:40]: just make this the dropped material, breaking sound, and material you use to fix variables, dont do this overriding stuff

File: code/game/objects/items/shields.dm [hunk 3]
```diff
@@ -67,58 +37,79 @@
...
-	transparent = FALSE
-	max_integrity = 55
-	w_class = WEIGHT_CLASS_NORMAL
+/obj/item/shield/riot
+	name = "riot shield"
+	desc = "A shield adept at blocking blunt objects from connecting with the torso of the shield wielder."
+	icon_state = "riot"
+	inhand_icon_state = "riot"
+	custom_materials = list(/datum/material/glass=7500, /datum/material/iron=1000)
+	var/cooldown = 0 //shield bash cooldown. based on world.time
```
Comment (Fikou) [inline code/game/objects/items/shields.dm:89]: use cooldown defines. new vars go on bottom

**Review by Fikou — COMMENTED** [review]
how does this interact with energy shields?

**Comment by GuillaumePrata** [comment]
The tests also made me realize that you can break adamantine shields now so I will refactor what energy shields use to be indestructible by damage into something easier to give shields without overwriting procs.

**Comment by ATH1909** [comment]
can you now fix a wooden shield by slapping it with wood instead?

**Comment by GuillaumePrata** [comment]
> can you now fix a wooden shield by slapping it with wood instead?

Nope, I want to remove this feature from the riot shields themselves so allowing wood shields to be repaired with wood just to open a PR right after this one that revert that would be shitty to the players.

**Review by Twaticus — APPROVED** [review]

**Comment by Mothblocks** [comment]
Conflicts

**Pull request merged**